### PR TITLE
Add custom eq_html RSpec matcher

### LINKED ISSUE #27: Create a custom RSpec matcher for component content
In our current project, I use to have this kind of tests.

```rb
      expect(component.to_html.tr("\n", "")).to eq <<~HTML.tr("\n", "").strip
        <label class="form-checkbox" for="user_is_enabled">
        <input name="user[is_enabled]" type="hidden" value="0">
        <input class="form-checkbox__input" type="checkbox" value="1" name="user[is_enabled]" id="user_is_enabled">
        <span class="form-checkbox__label">Actif</span>
        </label>
      HTML
```

It's not the cleanest. Maybe we may create a custom matcher like
```rb
      expect(component).to have_html <<~HTML
        <label class="form-checkbox" for="user_is_enabled">
        <input name="user[is_enabled]" type="hidden" value="0">
        <input class="form-checkbox__input" type="checkbox" value="1" name="user[is_enabled]" id="user_is_enabled">
        <span class="form-checkbox__label">Actif</span>
        </label>
      HTML
```

_Originally posted by @nicolas-brousse in https://github.com/pantographe/view_component-form/pull/14#discussion_r661302066_

### PR DESCRIPTION
**Pull request by Spone**
Close #27

### CODE REVIEW

**Comment by nicolas-brousse** [comment]
@Spone I permit myself to refactor a bit the matcher. What do you think about?

**Comment by Spone** [comment]
I prefer the output of my version:

image

It's more readable and easier to find the error with the diff, if you compare to your version:

image

**Comment by Spone** [comment]
But I like this change:

```diff
- expect(component.to_html).to eq_html <<~HTML
+ expect(component).to eq_html <<~HTML
```

File: spec/support/matchers/eq_html.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      @actual = nil
+      @invalid_response = nil
+    end
+
+    def matches?(html_fragment)
+      html_fragment = html_fragment.to_html if html_fragment.is_a? Nokogiri::HTML::DocumentFragment
+
+      @actual = html_fragment
+
+      @actual.include?(expected_formatted)
```
Comment (nicolas-brousse) [inline spec/support/matchers/eq_html.rb:18]:
```suggestion
      @actual == expected_formatted
```

File: spec/support/matchers/eq_html.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    end
+
+    def failure_message
+      "expected that HTML fragment contain #{expected_formatted}"
```
Comment (nicolas-brousse) [inline spec/support/matchers/eq_html.rb:22]: Take previous failure message back

**Review by nicolas-brousse — CHANGES_REQUESTED** [review]
Verify there is no new specs that must include this new matcher

**Review by nicolas-brousse — APPROVED** [review]

**Pull request merged**